### PR TITLE
Remove Invalid Tokens from cache when found

### PR DESCRIPTION
**Pull request by saquino0827**
# Description

- Uncache our TI token if an InvalidToken is caught and found.

## Issue
#1326 

## Checklist

- [ ] I have added tests to cover my changes
- [ ] I have added logging where useful (with appropriate log level)
- [ ] I have added JavaDocs where required
- [ ] I have updated the documentation accordingly

**Note**: You may remove items that are not applicable

### CODE REVIEW

File: app/src/main/java/gov/hhs/cdc/trustedintermediary/auth/AuthRequestValidator.java [hunk 1]
```diff
@@ -21,10 +21,12 @@ public class AuthRequestValidator {
     private static final AuthRequestValidator INSTANCE = new AuthRequestValidator();
 
     @Inject private AuthEngine jwtEngine;
-    @Inject private Cache keyCache;
+    @Inject Cache keyCache;
     @Inject private Secrets secrets;
     @Inject private Logger logger;
 
+    String ourPublicKey = "trusted-intermediary-public-key-" + ApplicationContext.getEnvironment();
```
Comment (jcrichlake) [inline app/src/main/java/gov/hhs/cdc/trustedintermediary/auth/AuthRequestValidator.java:28]: Do we want this public key name hardcoded here? And if so do we need a #pragma allow secret

Comment (saquino0827) [reply app/src/main/java/gov/hhs/cdc/trustedintermediary/auth/AuthRequestValidator.java:28]: It previously existed, but I moved it to a higher level so it's accessible in tests. I can add the #pragma allow secrets if we want but I also don't think it got flagged before as a secret.

Comment (jcrichlake) [reply app/src/main/java/gov/hhs/cdc/trustedintermediary/auth/AuthRequestValidator.java:28]: Cool

**Review by jcrichlake — APPROVED** [review]

File: app/src/main/java/gov/hhs/cdc/trustedintermediary/auth/AuthRequestValidator.java [hunk 2]
```diff
@@ -21,10 +21,12 @@ public class AuthRequestValidator {
     private static final AuthRequestValidator INSTANCE = new AuthRequestValidator();
 
     @Inject private AuthEngine jwtEngine;
-    @Inject private Cache keyCache;
+    @Inject Cache keyCache;
```
Comment (luis-pabon-tf) [inline app/src/main/java/gov/hhs/cdc/trustedintermediary/auth/AuthRequestValidator.java:24]: Seems this is the only inject we actually use in this class. Would a small doc be good?

Comment (halprin) [reply app/src/main/java/gov/hhs/cdc/trustedintermediary/auth/AuthRequestValidator.java:24]: Are you sure?  I see `jwtEngine`, `secrets`, and `logger` being used below.

Comment (saquino0827) [reply app/src/main/java/gov/hhs/cdc/trustedintermediary/auth/AuthRequestValidator.java:25]: I added some extra comments on the class. I think all the injects are used unless I'm misunderstanding what you mean.

**Review by pluckyswan — COMMENTED** [review]
Checklist needs to be updated (remove items or check them off)

**Pull request merged**